internal/manifest/oci_index.go

```diff
@@ -103,6 +116,7 @@ func (index *OCI1IndexPublic) editInstances(editInstances []ListEdit) error {
 			}
 			index.Manifests[targetIndex].MediaType = editInstance.UpdateMediaType
 		case ListOpAdd:
+			addCompressionAnnotations(editInstance.AddCompressionAnnotation, editInstance.AddAnnotations)
```

Comment:
Maps are reference types in Go (as shown in the `addCompressionAnnotations` parameter design); doesn’t this modify caller’s data?

Consider
```go
annotations := map[…]…
edits := []ListEdit{{…, annotations}, {…, annotations}}
```

AFAICS this would mean that if the first image adds an annotation, the second one would get it as well.